README.md

```diff
@@ -2,114 +2,121 @@
 
 # Goth
 
+<!-- MDOC !-->
+
 Google + Auth = Goth
 
 A simple library to generate and retrieve OAuth2 tokens for use with Google Cloud Service accounts.
 
-It can either retrieve tokens using service account credentials or from Google's metadata service for applications running on Google Cloud Platform.
-
 ## Installation
 
 1. Add Goth to your list of dependencies in `mix.exs`:
-  ```elixir
-  def deps do
-    [{:goth, "~> 1.2.0"}]
-  end
-  ```
-
-2. Pass in your credentials json downloaded from your GCE account:
-
-  ```elixir
-  config :goth,
-    json: "path/to/google/json/creds.json" |> File.read!
-  ```
-
-  Or, via an ENV var:
-  ```elixir
-  config :goth, json: {:system, "GCP_CREDENTIALS"}
-  ```
-
-  Or, via your own config module:
-  ```elixir
-  config :goth, config_module: MyConfigMod
-  ```
-  ```elixir
-  defmodule MyConfigMod do
-    use Goth.Config
-
-    def init(config) do
-      {:ok, Keyword.put(config, :json, System.get_env("MY_GCP_JSON_CREDENTIALS"))}
-    end
-  end
-  ```
-
-You can also use a JSON file containing an array of service accounts to be able to use different identities in your application. Each service
-account will be identified by its ```client_email```, which can be passed to ```Goth.Token.for_scope/1``` to specify which service account to use.
-
-For example, if your JSON file contains the following:
-
-```json
-[
-  {
-    "client_email": "account1@myproject.iam.gserviceaccount.com",
-    ...
-  },
-  {
-    "client_email": "account2@myproject.iam.gserviceaccount.com",
-    ...
-  }
-]
-```
 
-You can use the following to get a token for the second service account:
+   ```elixir
+   def deps do
+     [{:goth, "~> 1.3"}]
+   end
+   ```
+
+2. Add Goth to your supervision tree:
+
+   ```elixir
+   defmodule MyApp.Application do
+     use Application
+
+     def start(_type, _args) do
+       credentials = "GOOGLE_APPLICATION_CREDENTIALS_JSON" |> System.fetch_env!() |> Jason.decode!()
+
+       children = [
+         {Goth, name: MyApp.Goth, credentials: credentials}
+       ]
+
+       Supervisor.start_link(children, strategy: :one_for_one)
+     end
+   end
+   ```
+
+3. Fetch the token:
+
+    ```elixir
+    iex> {:ok, token} = Goth.fetch(MyApp.Goth)
+    iex> token
+    %Goth.Token{
+      expires: 1453356568,
+      token: "ya29.cALlJ4ICWRvMkYB-WsAR-CZnExE459PA7QPqKg5nei9y2T9-iqmbcgxq8XrTATNn_BPim",
+      type: "Bearer"
+    }
+    ```
+
+<!-- MDOC !-->
+
+## Upgrading from Goth < 1.3
+
+Earlier versions of Goth relied on global application environment configuration which is deprecated
+in favour of a more direct and explicit approach in Goth v1.3+.
+
+You might have code similar to this:
+
+```elixir
+# config/config.exs
+config :goth,
+  json: {:system, "GCP_CREDENTIALS"}
```

Comment:
what about the idea; instead of passing the variable name, it could digest pure value and let allow the client to configure `System.fetch_env!/1` on hiw own

Reply:
ah, you're right! then, on the same page, thanks